src/query.rs

```diff
@@ -234,6 +239,96 @@ impl From<Vec<Arc<OverrideMap>>> for OverrideStack {
     }
 }
 
+/// Data for generating a **witness** from the results of a [`SemverQuery`].
+///
+/// A witness is a minimal compilable example of how downstream code would
+/// break given this change.  See field documentation for more information
+/// on each member.
+///
+/// Fields besides [`hint_template`](Self::hint_template) are optional, as it is not
+/// always necessary to use an additional query [`witness_query`](Self::witness_query)
+/// or possible to build a compilable witness from [`witness_template`](Self::witness_template)
+/// for a given `SemverQuery`.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Witness {
+    /// A [`handlebars`] template that renders a user-facing hint to give a quick
+    /// explanation of breakage.  This may not be a buildable example, but it should
+    /// show the idea of why downstream code could break.  It will be provided any
+    /// `@output` query declarations in the [`SemverQuery`] query.
+    ///
+    /// Example for the `function_missing` lint, where `name` is the (re)moved function's
+    /// name and `path` is the importable path:
+    ///
+    /// ```no_run
+    /// # let _ = r#"
+    /// "use {{join "::" path}};
+    /// {{name}}(...);
+    /// # "#;
+    /// ```
+    ///
+    /// Notice how this is not a compilable example, but it provides a distilled hint to the user
+    /// of how downstream code would break with this change.
+    pub hint_template: String,
+
+    /// A [`handlebars`] template that renders the compilable witness example of how
+    /// downstream code would break.
+    ///
+    /// This template will be provided any fields with `@output` directives in the
+    /// original [`SemverQuery`].  If [`witness_query`](Self::witness_query) is `Some`,
+    /// it will also be provided the `@output`s of that query. (The additional query's
+    /// outputs will take precedence over the original query if they share the same name.)
+    ///
+    /// Example for the `enum_variant_missing` lint, where `path` is the importable path of the enum,
+    /// `name` is the name of the enum, and `variant_name` is the name of the removed/renamed variant:
+    ///
+    /// ```no_run
+    /// # let _ = r#"
+    /// use {{join "::" path}};
+    /// fn witness(item: {{name}}) {
+    ///     if let {{name}}::{{variant_name}}{..} = item {
+    ///
+    ///     }
+    /// }
+    /// # "#;
```

Comment:
This is a very reasonable but unfortunately incorrect witness. This stuff is hard!

The issue is that `name` is the _declared_ name of the enum, but it might be renamed as part of the importable path. For example:
```rust
mod internal {
    pub enum Example {
        Variant,
    }
}

pub use internal::Example as PubExample;
```

If we removed `Variant` here, the generated witness will say:
```rust
use my_crate::PubExample;
fn witness(item: Example) {
    if let Example::Variant{..} = item {

    }
}
```
This won't compile since `Example` is undefined.

The easiest fix is to always use the fully-qualified path instead of importing it.
```suggestion
    /// # let _ = r#"
    /// fn witness(item: {{path}}) {
    ///     if let {{path}}::{{variant_name}} {..} = item {
    ///
    ///     }
    /// }
    /// # "#;
```